src/main/java/com/battleships/battleships/BattleshipSquare.java

```diff
@@ -1,13 +1,17 @@
 package com.battleships.battleships;
 
+//inheritance versus aggregation?
```

Comment:
I think this is the crux of what I'm struggling with.  You seem to have conflated the notion of a grid square and a ship.  The way you've modelled it is with a ship *as-a* type of `BattleshipSquare` (i.e. inheritance).  However to me it feels more natural to say a ship *has-a* collection of `BattleshipSquares` (i.e. aggregation).
The result is that you end up using instances of `BattleshipSquare` in two different ways, as a ship and as a square, hence you have fields which don't make much sense for some of the instances.